### PR TITLE
[SP-5777] - Rollback and service scale playbooks

### PR DESCRIPTION
**Pull request by hagaishapira**
### General
- [ ] Go over texts with @stackpulse/product
- [ ] Explicitly discuss the _playbook name_, _image_ and _badges_
- [ ] Explicitly consider various _benefits_. As a rule of thumb, there should be 2-3 of them
- [ ] If the playbook has Slack outputs use the Slack experience template

### New playbooks
- [ ] Tested Playbook in a production tenant
- [ ] Added a link in the [README.MD](https://github.com/stackpulse/playbooks/blob/master/README.md)

### Playbook structure
- [ ] For improved human-readability, please add a new empty line between the steps
- [ ] Please add a one-liner comment block before each step, for example: `# Retrieve long connections from the database server`
- [ ] When considering step IDs, please use short, but descriptive terms, explaining what is being done, such as `retrieve_connections_data` or `update_storage_size`
- [ ] When writing params descriptions, start with a capital letter and phrase as a short sentence, such as `User name for database connection` or `Address of the database cluster`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,24 +19,48 @@ Playbooks in this section enrich, analyze and triage alerts in real-time. They h
...
+      <a href="https://github.com/stackpulse/playbooks/tree/master/kubernetes/service-scale">
       <img src="https://raw.githubusercontent.com/kubernetes/kubernetes/master/logo/logo.png"/>
       </a>
     </td>
     <td border=0 width="90%">
-      <a href="https://github.com/stackpulse/playbooks/tree/master/kubernetes/pod-restarting"><b>Kubernetes Pod Restarting</b></a><br><br>
-     This playbook solves consistent Pod Restarting events in a Kubernetes cluster.
-     It gets the latest started pods in the namespace provided either by alert or by the user, then gets the current and previous (if exists) logs of the relevant container.
+      <a href="https://github.com/stackpulse/playbooks/tree/master/kubernetes/service-scale"><b>Kubernetes Service Scale</b></a><br><br>
+      This playbook enables an operator to easily and properly rollback a Kubernetes deployment using a GitOps Flux CD pipeline. Executing the playbook will interactively retrieve the list of services to the user and then the commit history for the selected service. Once the correct target tag is chosen, the playbook continues to safely rollback the service and lock the CD pipeline.
```
Comment (vic3lord) [inline README.md:55]:
```suggestion
      This playbook enables an operator to easily and properly rollback a Kubernetes deployment using a GitOps FluxCD. Executing the playbook will interactively retrieve the list of services to the user and then the commit history for the selected service. Once the correct target tag is chosen, the playbook continues to safely rollback the service and lock the CD pipeline.
```

File: kubernetes/rollback/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png" width="100">
+
+![env](https://img.shields.io/static/v1?label=env&message=Kubernetes&style=flat&logo=Kubernetes&color=326CE5)
+![uses-slack](https://img.shields.io/static/v1?label=uses&message=Slack&style=flat&logo=slack&color=4A154B)
+![uses-github](https://img.shields.io/static/v1?label=uses&message=GitHub&style=flat&logo=github&color=4078c0)
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/rollback/playbook.yaml)
+
+This playbook enables an operator to easily and properly rollback a Kubernetes deployment using a GitOps Flux CD pipeline. Executing the playbook will interactively retrieve the list of services to the user and then the commit history for the selected service. Once the correct target tag is chosen, the playbook continues to safely rollback the service and lock the CD pipeline.
```
Comment (vic3lord) [inline kubernetes/rollback/README.md:11]:
```suggestion
This playbook enables an operator to easily and properly rollback a Kubernetes deployment using a GitOps FluxCD. Executing the playbook will interactively retrieve the list of services to the user and then the commit history for the selected service. Once the correct target tag is chosen, the playbook continues to safely rollback the service and lock the CD pipeline.
```

File: kubernetes/rollback/playbook.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    type: var
+    description: Last n version tags
+    default: "5"
+  - name: user
+    type: var
+    description: User performing the rollback (for Github message purposes)
+  - name: targetClusterRunner
+    type: spd
+    description: SPD runner on destination cluster.
+  # Slack
```
Comment (eldadru) [inline kubernetes/rollback/playbook.yaml:20]: Do we need this comment?

File: kubernetes/rollback/playbook.yaml [hunk 4]
```diff
@@ -0,0 +1,120 @@
+apiVersion: stackpulse.io/v1
+kind: Playbook
+metadata:
+  name: deployment-rollback
+  description: Update K8s deployment with a new image based on Github tag
+parameters:
+  - name: recipient
+    type: var
+    description: Slack recipient user or channel name
```
Comment (eldadru) [inline kubernetes/rollback/playbook.yaml:9]: Recipient of what? I think the description should state what we are going to send that destination

File: kubernetes/rollback/playbook.yaml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  - name: recent_tags_count
+    type: var
+    description: Last n version tags
```
Comment (eldadru) [inline kubernetes/rollback/playbook.yaml:12]:
```suggestion
    description: Amount of GitHub tags to display as candidates for the rollback operation.
```

File: kubernetes/rollback/playbook.yaml [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    default: "5"
+  - name: user
+    type: var
+    description: User performing the rollback (for Github message purposes)
```
Comment (eldadru) [inline kubernetes/rollback/playbook.yaml:16]: Email? username? "user" is too open - is that "user_email"? "user_github_handle"

File: kubernetes/rollback/playbook.yaml [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  - name: targetClusterRunner
+    type: spd
+    description: SPD runner on destination cluster.
+  # Slack
+  - name: slackIntegrationName
+    type: integration
+    description: Slack integration name.
```
Comment (eldadru) [inline kubernetes/rollback/playbook.yaml:23]:
```suggestion
    description: Slack integration to use when sending slack messages.
```

File: kubernetes/rollback/playbook.yaml [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+      RESPONSES: "{{ toJson $.get_tags.Tags.[:].Name }}"
+    envFrom:
+      integrationRef: "{{ $.params.slackIntegrationName }}"
+  - name: us-docker.pkg.dev/stackpulse/public/fluxcd/lock
+    runner: "{{ $.params.targetClusterRunner }}"
+    id: flux_lock
+    env:
+      NAME: "{{ $.slack_send_interactive_message.selected_responses[0] }}"
+      NAMESPACE: default
+      FLUX_URL: http://flux.flux:3030/api/flux
```
Comment (eldadru) [inline kubernetes/rollback/playbook.yaml:88]: param? and that value as the default?

File: kubernetes/rollback/playbook.yaml [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+      HEADERS: '{"age":"Age", "name":"Name", "status":"Status"}'
+      INCLUDE_FILTER: '["age", "name", "status"]'
+  - name: us-docker.pkg.dev/stackpulse/public/slack/message/snippet
+    id: slack_send_pods_snippet
+    env:
+      CONTENT: "{{ $.tabulate_pods.tabulate_output }}"
+      RECIPIENTS: "{{ $.params.recipient }}"
+      SP_REDIRECT_URL: "{{ $.metadata.executionUrl }}"
+      FILE_TYPE: auto
+      TITLE: "Rollback completed {{ $.slack_send_interactive_message.selected_responses[0] }} running pods:"
```
Comment (eldadru) [inline kubernetes/rollback/playbook.yaml:118]: Why the ":"? it doesn't seem we are displaying anything below.

File: kubernetes/service-scale/playbook.yaml [hunk 10]
```diff
@@ -0,0 +1,143 @@
+apiVersion: stackpulse.io/v1
+kind: Playbook
+metadata:
+  name: service-scale
+  description: This playbook will edit the range of a service HPA (Horizontal Pod Autoscaler).
+parameters:
+  # Operation Subject
```
Comment (eldadru) [inline kubernetes/service-scale/playbook.yaml:7]: comment is redundant IMO

File: kubernetes/service-scale/playbook.yaml [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+    type: spd
+    description: SPD runner on destination cluster.
+  - name: targetMinReplicas
+    type: var
+    description: The minimum number of replicas to use.
+  - name: targetMaxReplicas
+    type: var
+    description: The maximum number of replicas to use.
+
+  # Github
```
Comment (eldadru) [inline kubernetes/service-scale/playbook.yaml:25]: the comment is redundant IMO

File: kubernetes/rollback/playbook.yaml (hunk 9 above)
Comment (hagaishapira) [reply kubernetes/rollback/playbook.yaml:118]: It has a snippet

**Review by vic3lord — APPROVED** [review]

**Pull request merged**